coloc/coloc_runner.py

```diff
@@ -0,0 +1,246 @@
+
+    with (ro.default_converter + pandas2ri.converter).context():
+        gwas_r = ro.conversion.get_conversion().py2rpy(gwas)
+    ro.globalenv['gwas_r'] = gwas_r
+    ro.r(
+        '''
+    gwas_r = gwas_r %>% select(beta, varbeta, position,snp)
+    gwas_r = gwas_r %>% filter((beta!=0) | (varbeta!=0))
+    gwas_r = gwas_r %>% distinct(snp, .keep_all = TRUE)
+    gwas_r = gwas_r%>% as.list()
+    gwas_r$type = 'cc'
+
+    ''',
+    )
+    eqtl = pd.read_csv(
+        eqtl_file_path,
+        sep='\t',
+    )
+    eqtl['beta'] = eqtl['BETA']
+    eqtl['se'] = eqtl['SE']
+    eqtl['position'] = eqtl['POS']
+    eqtl['snp'] = eqtl['MarkerID'].str.replace(':', '_', regex=False)
+    gene = eqtl_file_path.split('/')[-1].replace(f'{celltype}_', '').replace('_cis', '')
+    with (ro.default_converter + pandas2ri.converter).context():
+        eqtl_r = ro.conversion.get_conversion().py2rpy(eqtl)
+    ro.globalenv['eqtl_r'] = eqtl_r
+    ro.globalenv['gene'] = gene
+    ro.r(
+        '''
+    eqtl_r = eqtl_r %>% filter(!is.na(beta))
+    eqtl_r = eqtl_r %>% distinct(snp, .keep_all = TRUE)
+    eqtl_r$varbeta = eqtl_r$se**2
+    eqtl_r$position = eqtl_r$pos
+    eqtl_r = eqtl_r %>% select(beta, varbeta, position, snp)
+
+    eqtl_r = eqtl_r %>% as.list()
+    eqtl_r$type = 'quant'
+    eqtl_r$sdY = 1
+
+
+    my.res <- coloc.abf(dataset1=gwas_r,
+                    dataset2=eqtl_r)
+
+    p_df <- data.frame(gene,my.res$summary[1], my.res$summary[2], my.res$summary[3], my.res$summary[4], my.res$summary[5], my.res$summary[6])
+    names(p_df) <- c('gene', 'nsnps_coloc_tested','PP.H0.abf','PP.H1.abf','PP.H2.abf','PP.H3.abf','PP.H4.abf')
+    ''',
+    )
+
+    # convert to pandas df
+    with (ro.default_converter + pandas2ri.converter).context():
+        pd_p4_df = ro.conversion.get_conversion().rpy2py(ro.r('p_df'))
+
+    # add cell type and chrom annotation to df
+    pd_p4_df['celltype'] = celltype
+    pd_p4_df['chrom'] = eqtl['chr'].iloc[0]
+
+    # write to GCS
+    pd_p4_df.to_csv(
+        coloc_results_file,
+        sep='\t',
+        index=False,
+    )
+
+
+@click.option(
+    '--egenes-files-path',
+    help='Path to the gene-level summary files',
+    default='gs://cpg-bioheart-test-analysis/saige-qtl/bioheart_n990_and_tob_n1055/241004_n100/output_files/summary_stats',
+)
+@click.option(
+    '--snp-cis-dir',
+    help='Path to the directory containing the SNP cis results',
+    default='gs://cpg-bioheart-test-analysis/saige-qtl/bioheart_n990_and_tob_n1055/241004_n100/output_files',
+)
+@click.option(
+    '--snp-gwas-file',
+    help='Path to the SNP GWAS file',
+    default='gs://cpg-bioheart-test/str/gwas_catalog/gcst/gcst-gwas-catalogs/GCST011071_parsed.tsv',
+)
+@click.option(
+    '--celltypes', help='Cell types to run, single str, comma separated', default='ASDC'
+)
+@click.option('--cis-window-size', help='Cis window size used', default=100000)
+@click.option('fdr-threshold', help='FDR threshold', default=0.05)
```

Comment:
```suggestion
@click.option('--fdr-threshold', help='FDR threshold', default=0.05)
```

Reply:
ah oops sorry didn't see this! Changed that below, would have saved me a couple of minutes 😅